### PR TITLE
MBL-10246 show locked assignment details

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-10246
affects: Student
release note: none

### CODE REVIEW

**Comment by zachsis** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-10246](https://instructure.atlassian.net/browse/MBL-10246)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.17% | 95.17% | 0%
**Student** | 92.89% | 92.89% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.69% | 92.69% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Assignments/AssignmentDetails/AssignmentDetailsViewController.swift [hunk 1]
```diff
@@ -247,6 +256,45 @@ class AssignmentDetailsViewController: UIViewController, AssignmentDetailsViewPr
...
+            lockedSection?.isHidden = false
+            fileTypesSection?.isHidden = true
+            submissionTypesSection?.isHidden = true
+            gradeSection?.isHidden = true
+            submissionButtonSection?.isHidden = true
+            showDescription(false)
+            submitAssignmentButton.isHidden = true
+        case .after:
+            lockedSection?.isHidden = false
+            submitAssignmentButton.isHidden = true
```
Comment (armstrongnate) [inline Student/Student/Assignments/AssignmentDetails/AssignmentDetailsViewController.swift]: Should this case also update all the views in `.before` in the scenario where the `lockStatus` changes in a pull to refresh?

Comment (bluwave) [reply Student/Student/Assignments/AssignmentDetails/AssignmentDetailsViewController.swift]: yeah, good point

File: Student/StudentUnitTests/Assignments/AssignmentDetails/AssignmentDetailsPresenterTests.swift [hunk 2]
```diff
@@ -190,6 +190,174 @@ class AssignmentDetailsPresenterTests: PersistenceTestCase {
...
+        XCTAssertFalse( presenter.descriptionIsHidden() )
+    }
+
+    func testDescriptionSectionNotHidden() {
+        setupIsHiddenTest(lockStatus: .unlocked)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.descriptionIsHidden() )
+    }
+
+    func testsubmitAssignmentButtonIsHiddenBeforeAvailability() {
```
Comment (armstrongnate) [inline Student/StudentUnitTests/Assignments/AssignmentDetails/AssignmentDetailsPresenterTests.swift]: These last 3 tests need capital `S` to get picked up by the test runner. `testSubmit...`

File: Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift [hunk 3]
```diff
@@ -150,4 +150,46 @@ class AssignmentDetailsPresenter {
...
+        return assignment?.lockStatus == .before
+    }
+
+    func descriptionIsHidden() -> Bool {
+        return assignment?.lockStatus == .before
+    }
+
+    func submitAssignmentButtonIsHidden() -> Bool {
+        return assignment?.lockStatus != .unlocked
+    }
```
Comment (armstrongnate) [inline Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift]: I like these, very nice.

File: Student/StudentUnitTests/Assignments/AssignmentDetails/AssignmentDetailsPresenterTests.swift (hunk 2 above)
Comment (bluwave) [reply Student/StudentUnitTests/Assignments/AssignmentDetails/AssignmentDetailsPresenterTests.swift]: doh

**Pull request merged**